src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -764,15 +763,169 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
             assert(op1 != nullptr);
             assert(op2 != nullptr);
             assert(baseType == TYP_DOUBLE);
-
-            ins    = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
-            op2Reg = op2->gtRegNum;
-            ival   = Compiler::ivalOfHWIntrinsic(intrinsicID);
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            op2Reg          = op2->gtRegNum;
+            ival            = Compiler::ivalOfHWIntrinsic(intrinsicID);
             emit->emitIns_SIMD_R_R_R_I(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg, ival);
 
             break;
         }
 
+        case NI_SSE2_CompareEqualOrderedScalar:
```

Comment:
It would be good to have an explicit bug that tracks merging this with the SSE implementation.

Reply:
The issue #16330 tracks this work item. Could you assign it to me within hardware intrinsics project?

Reply:
Done.